fearless/Modules/Staking/StakingMain/StakingMainInterator+Subscription.swift

```diff
@@ -314,204 +130,171 @@ extension StakingMainInteractor {
-        controllerAccountProvider = nil
-    }
-
-    func subscribeToPayee(address: String) {
-        guard payeeProvider == nil else {
-            return
+    func handleLedgerInfo(
+        result: Result<StakingLedger?, Error>,
+        accountId _: AccountId,
+        chainId _: ChainModel.Id
+    ) {
+        switch result {
+        case let .success(ledgerInfo):
+            presenter.didReceive(ledgerInfo: ledgerInfo)
+        case let .failure(error):
+            presenter.didReceive(ledgerInfoError: error)
         }
+    }
 
-        guard let payeeProvider = try? providerFactory
-            .getPayee(for: address, runtimeService: runtimeService)
-        else {
-            logger.error("Can't create payee provider")
-            return
+    func handleNomination(
+        result: Result<Nomination?, Error>,
+        accountId _: AccountId,
+        chainId _: ChainModel.Id
+    ) {
+        switch result {
+        case let .success(nomination):
+            presenter.didReceive(nomination: nomination)
+        case let .failure(error):
+            presenter.didReceive(nominationError: error)
         }
+    }
 
-        self.payeeProvider = payeeProvider
-
-        let updateClosure = { [weak self] (changes: [DataProviderChange<DecodedPayee>]) in
-            if let payee = changes.reduceToLastChange() {
-                self?.presenter.didReceive(payee: payee.item)
-            }
+    func handleValidator(
+        result: Result<ValidatorPrefs?, Error>,
+        accountId _: AccountId,
+        chainId _: ChainModel.Id
+    ) {
+        switch result {
+        case let .success(validatorPrefs):
+            presenter.didReceive(validatorPrefs: validatorPrefs)
+        case let .failure(error):
+            presenter.didReceive(validatorError: error)
         }
+    }
 
-        let failureClosure = { [weak self] (error: Error) in
-            self?.presenter.didReceive(payeeError: error)
-            return
+    func handlePayee(
+        result: Result<RewardDestinationArg?, Error>,
+        accountId _: AccountId,
+        chainId _: ChainModel.Id
+    ) {
+        switch result {
+        case let .success(payee):
+            presenter.didReceive(payee: payee)
+        case let .failure(error):
+            presenter.didReceive(payeeError: error)
         }
-
-        let options = DataProviderObserverOptions(
-            alwaysNotifyOnRefresh: false,
-            waitsInProgressSyncOnAdd: false
-        )
-        payeeProvider.addObserver(
-            self,
-            deliverOn: .main,
-            executing: updateClosure,
-            failing: failureClosure,
-            options: options
-        )
     }
 
-    func subscribeToControllerAccount(address: AccountAddress) {
-        guard controllerAccountProvider == nil else {
+    func handleTotalReward(
+        result: Result<TotalRewardItem, Error>,
+        for address: AccountAddress,
+        api _: ChainModel.ExternalApi
```

Comment:
Could you please clarify why we need to pass api as an argument to a handling function?